From 08d70c87b1e76f85537645f387631c65c7e70a4e Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Fri, 9 Jul 2021 15:35:50 +0200
Subject: [PATCH 1/4] usb: dwc3: Disable hibernation by default

Hibernation support appears to be badly broken, plugging in a cable causes kernel panics like:
  BUG: scheduling while atomic: irq/50-dwc3/403/0x00010100

Disabling hibernation support works around it. Can be enabled through debugfs though.

Fixes: 511e872607db "usb: dwc3: gadget: Add hibernation support when operating in gadget mode"
---
 drivers/usb/dwc3/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 12ccaf6f76e0..a8996a651279 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -959,7 +959,7 @@ static void dwc3_core_setup_global_control(struct dwc3 *dwc)
 	case DWC3_GHWPARAMS1_EN_PWROPT_HIB:
 		/* enable hibernation here */
 		dwc->nr_scratch = DWC3_GHWPARAMS4_HIBER_SCRATCHBUFS(hwparams4);
-		dwc->has_hibernation = 1;
+		dwc->has_hibernation = 0;
 
 		/*
 		 * REVISIT Enabling this bit so that host-mode hibernation
-- 
2.17.1

